### PR TITLE
update polly and update how we use the API

### PR DESCRIPTION
**Pull request by akiraveliara**
this might also address some performance issues with retrying reported. i'll look into these later.

### CODE REVIEW

File: DSharpPlus/Net/Rest/RateLimitStrategy.cs [hunk 1]
```diff
@@ -16,64 +16,38 @@ namespace DSharpPlus.Net;
 
 using System.Collections.Generic;
 
-internal class RateLimitPolicy : AsyncPolicy<HttpResponseMessage>
+internal class RateLimitStrategy(ILogger logger, int waitingForHashMilliseconds = 200) : ResilienceStrategy<HttpResponseMessage>
```
Comment (Plerx2493) [inline DSharpPlus/Net/Rest/RateLimitStrategy.cs:19]: Dont know about primary constructors, personaly i find them pretty confusing in bigger classes but only a nit

File: DSharpPlus/Net/Rest/RateLimitStrategy.cs [hunk 2]
```diff
@@ -152,40 +126,48 @@ public RateLimitPolicy(ILogger logger, int waitingForHashMilliseconds = 200)
...
             
             this.waitingForHashListSemaphore.Release();
         }
 
         // make the actual request
 
-        HttpResponseMessage response = await action(context, cancellationToken);
+        Outcome<HttpResponseMessage> outcome = await action(context, state);
+        
+        if (outcome.Result is null)
```
Comment (Plerx2493) [inline DSharpPlus/Net/Rest/RateLimitStrategy.cs:156]: I asume checking the result for null is the same as "is successful"?

**Review by Plerx2493 — COMMENTED** [review]
Looks good, had some question and provided an answer :3

File: DSharpPlus/Net/Rest/RateLimitStrategy.cs (hunk 2 above)
Comment (akiraveliara) [reply DSharpPlus/Net/Rest/RateLimitStrategy.cs:156]: yes

**Review by Plerx2493 — APPROVED** [review]
LGTM, will wait to merge until we discussed the primary constructor thing in LibDev

**Pull request merged**